src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,282 @@
+    10
+  );
+
+  const euiSizeMediumNumber: number = parseInt(
+    euiTheme.size.m.replace('px', ''),
+    10
+  );
+
+  const flyoutSizes = {
+    s: {
+      min: Math.round(euiTheme.breakpoint.m * 0.5),
+      width: '25vw',
+      max: Math.round(euiTheme.breakpoint.s * 0.7),
+    },
+
+    m: {
+      min: euiFormMaxWidthNumber + euiSizeMediumNumber * 2,
+      width: '50vw',
+      max: euiTheme.breakpoint.m,
+    },
+
+    l: {
+      min: Math.round(euiTheme.breakpoint.m * 0.9),
+      width: '75vw',
+      max: Math.round(euiTheme.breakpoint.l),
+    },
+  };
+
+  return {
+    euiFlyout: css`
+      ${logicalCSS('border-left', euiTheme.border.thin)}
+      ${euiShadowXLarge(euiThemeContext)};
+      position: fixed;
+      ${logicalCSS('top', 0)}
+      ${logicalCSS('right', 0)}
+      ${logicalCSS('bottom', 0)}
+      ${logicalCSS('height', '100%')}
+      z-index: ${euiTheme.levels.header};
+      background: ${euiTheme.colors.emptyShade};
+      display: flex;
+      flex-direction: column;
+      align-items: stretch;
+      clip-path: polygon(-50% 0, 100% 0, 100% 100%, -50% 100%);
+      ${euiCanAnimate} {
+        animation: ${euiFlyout} ${euiTheme.animation.normal}
+          ${euiTheme.animation.resistance};
+      }
+      &:focus {
+        outline: none;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('max-width', '90vw')}
+      }
+    `,
+
+    // Flyout Sizes
+    // Note: Dashes are used because s, m, l are size values for multiple props
+    'flyoutSize--s': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.s.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.s.min}px`)}
+        ${logicalCSS('width', flyoutSizes.s.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.s.min}px`)}
+      }
+    `,
+    'flyoutSize--m': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.m.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.m.min}px`)}
+        ${logicalCSS('width', flyoutSizes.m.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.m.min}px`)}
+      }
+    `,
+    'flyoutSize--l': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.l.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.l.min}px`)}
+        ${logicalCSS('width', flyoutSizes.l.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.l.min}px`)}
+      }
```

Comment:
Couple of notes:

1. Let's lose the `euiFlyout--maxWidth-default` modifier - it's not being set anymore

2. This CSS is pretty repetitive as well - why not make a DRY util for it, similar to the padding helpers?

```suggestion
const composeFlyoutSize = (euiThemeContext, size) => {
  // set up flyoutSizes map

  return css`
    ${logicalCSS('max-width', `${flyoutSizes.m.max}px`)}

    ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
      ${logicalCSS('min-width', `${flyoutSizes.m.min}px`)}
      ${logicalCSS('width', flyoutSizes.m.width)}
    }

    ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
      ${logicalCSS('min-width', 0)}
      ${logicalCSS('width', `${flyoutSizes.m.min}px`)}
    }
  `;
}

// usage 
flyoutSizes: {
  s: composeFlyoutSize('s'),
  m: composeFlyoutSize('m'),
  l: composeFlyoutSize('l'),
},
```